test/clj/puppetlabs/trapperkeeper/testutils/webserver.clj

```diff
@@ -55,3 +58,25 @@
      ~port-var
      {}
      ~@body))
+
+(defn get-jetty-mbean-object-names
+  "TODO SPLAIN"
+  []
+  (let [mbeanserver (ManagementFactory/getPlatformMBeanServer)]
+    (.queryNames mbeanserver
+                 nil
+                 (ObjectName. "org.eclipse.jetty.jmx:type=mbeancontainer,*"))))
+
+(defn assert-clean-shutdown
+  "TODO SPLAIN"
+  [f]
+  (f)
+  ;; This sucks, because if this assertion fails, it will not give any clue as to
+  ;; which test caused it to fail beyond just the test namespace.  However, I tried
+  ;; several things (such as throwing an exception here rather than having an assertion),
+  ;; and nothing gave any better debugging info because the metadata about which
+  ;; test we're running and the call stack for the test function itself are simply
+  ;; not available from inside a fixture.  So I decided that even those these aren't
+  ;; super fun to debug, it was better than any other option in terms of enforcing
+  ;; that the tests clean up after themselves properly.
```

Comment:
The mbean registry cleanup is a per-Jetty-server-instance thing, so if even a single test doesn't clean up, the assertion will fail at the end up of the namespace.  i.e. if one test doesn't clean up after its `Server` instance, a subsequent test that cleaned up after itself wouldn't clean up for the earlier test.